jenkins/bootstrap.py

```diff
@@ -727,33 +729,44 @@ def bootstrap(
     else:
         end = 0
     call = lambda *a, **kw: _call(end, *a, **kw)
+    gsutil = GSUtil(call)
+
     logging.info('Bootstrap %s...', job)
     build = build_name(started)
-    setup_root(call, root, repo, branch, pull, ssh)
-    logging.info('Configure environment...')
-    if repo:
-        version = find_version(call)
-    else:
-        version = ''
-    setup_magic_environment(job)
-    setup_credentials(call, robot, upload)
+
     if upload:
         if pull:
             paths = pr_paths(upload, repo, job, build, pull)
         else:
             paths = ci_paths(upload, job, build)
         logging.info('Gubernator results at %s', gubernator_uri(paths))
-    gsutil = GSUtil(call)
-    logging.info('Start %s at %s...', build, version)
-    if upload:
-        start(gsutil, paths, started, node(), version, pull)
-    success = False
+
+    version = 'unknown'
+    exc_type = None
+
     try:
-        call(job_script(job, use_json))
-        logging.info('PASS: %s', job)
-        success = True
-    except subprocess.CalledProcessError:
-        logging.error('FAIL: %s', job)
+        setup_root(call, root, repo, branch, pull, ssh)
+        logging.info('Configure environment...')
+        if repo:
+            version = find_version(call)
+        else:
+            version = ''
+        setup_magic_environment(job)
+        setup_credentials(call, robot, upload)
+        logging.info('Start %s at %s...', build, version)
+        if upload:
+            start(gsutil, paths, started, node(), version, pull)
```

Comment:
We may never call `start()` or `finish()`, thus we'll just have `build-log.txt`. Should we attempt to upload these files anyway?